

<html>
<body>
 <h1><b>Phylogenetic and evolutionary relationships among yellow fever virus isolates in Africa and South America </b></h1>
 <img src="http://www.historyofvaccines.org/sites/default/files/uploaded-timeline-overview-images/000425_265.jpg" alt="Mountain View" style="width:250px;height:25px;">
        <MARQUEE><img src= "https://microbewiki.kenyon.edu/images/thumb/2/25/YF_Virion.jpg/300px-YF_Virion.jpg"></MARQUEE>
        <a href="https://github.com/ebaberga" class="button"><small>Follow me on</small> GitHub</a>
      
     
   </header> 
<h2><b>Review of what I did in the Bedford lab:</b></h2>
<ol>
<h4> First and Second week<a href= "first.md"> HERE</a></h4>
<h4>Third and Fourth week<a href= "third.md"> HERE</a></h4>
<h4>Fifth week <a href= ""> HERE</a></h4>
</ol>
<h2>Project Question </h3>
<ol>
<li>How similar or Different is the yellow fever outbreaks in different regions of the world?</li>
<li>What can we tell about the host variation among this regions?</li>
<h3>Targeted regions of my project:</h2>
<ol>
<li>East Africa </li>
<li>West Africa </li>
<li>South America </li>
</ol>

<P><b>The reason I chose these regions is because they have redundant outbreaks of Yellow Fever since its discovery, 1927.</b></p>
<h2><b>Steps towards building a phylogenetic tree for Yellow Fever:</b></h2>
<p>All my sequence are draged from <a href="http://www.viprbrc.org/brc/vipr_genome_search.spg?method=SubmitForm&blockId=2713&decorator=flavi"</a> here </p>
<h3>I used the following application softwares throughout my project to do multiple tasks:</h3>
<ol>
<li> Jupyter Notebook: to build my subdata set.</li> <a href="img src=dataset.png"> here </a> 
<li> Geneious: to do my alignments using MAFT and draw a phylogenetic tree using RAFT of the sequences I filtered out using Jupyter Notebook (The first time I did it ).</li>
<li>Command line: to make my alignments for the 3rd time, and install most of the softwares I used so far.</li>
<li>Figtree: to graphically view my phylogenetic tree and publish ready figure.</li>
<li>Beauti: a companion program in Beast that generate XML file inorder to make the file readable by Beast. </li>
<li>Beast: to infer evolutionary dynamics from sequence data.</li>
</ol>
<h4><b>Why is these so important to me: </b></h4><p><b>I have a dream of pursuing a Computer Engineering degree in the near future. Using this software will broaden my understanding of computers in general, and make me a competitive college applicant.</b></p>
<h3><b>My final tree:</b> </h3>
<a href="img src="tree.jpg">here</a>
<h3>The tree demonstrates the variation of sampled hosts in East Africa, West Africa, and South America(Brazil and Venezuela):</h3>
<h4>Things I understood from my Phylogenetic tree:</h4>
<ul>
<li>Most mosquito samples are from African counties, while most human samples are from South America(Brazil and Venezuela).</li>
<li>The virus tends to stay in a specific region for a long time, it doesn't jump around different regions and hosts often. </li>
<li>The tree demonstrate that it is possible for Yellow Fever virus to jump from one species to another.</li>
</ul>
<h4>Figures and Hypothesis</h4><a href = "result.md">here</a>
<h4>What is next</h4><a href="next.md">here</a>
</html>

